java/client/src/test/java/glide/managers/CommandManagerTest.java

```diff
@@ -234,4 +240,112 @@ public void prepare_request_with_unknown_route_type() {
                         IllegalArgumentException.class, () -> service.submitNewCommand(command, r -> null));
         assertEquals("Unknown type of route", exception.getMessage());
     }
+
+    @SneakyThrows
+    @Test
+    public void submitNewCommand_with_Transaction_sends_protobuf_request() {
+        // setup
+        String[] arg1 = new String[] {"GETSTRING", "one"};
+        String[] arg2 = new String[] {"GETSTRING", "two"};
+        String[] arg3 = new String[] {"GETSTRING", "three"};
+        Transaction trans = new Transaction();
+        trans.customCommand(arg1).customCommand(arg2).customCommand(arg3);
+
+        CompletableFuture<Response> future = new CompletableFuture<>();
+        when(channelHandler.write(any(), anyBoolean())).thenReturn(future);
+
+        ArgumentCaptor<RedisRequest.Builder> captor =
+                ArgumentCaptor.forClass(RedisRequest.Builder.class);
+
+        // exercise
+        service.submitNewCommand(trans, r -> null);
+
+        // verify
+        verify(channelHandler).write(captor.capture(), anyBoolean());
+        var requestBuilder = captor.getValue();
+
+        // verify
+        assertTrue(requestBuilder.hasTransaction());
+        assertEquals(3, requestBuilder.getTransaction().getCommandsCount());
+
+        LinkedList<String> resultPayloads = new LinkedList<>();
+        resultPayloads.add("one");
+        resultPayloads.add("two");
+        resultPayloads.add("three");
+        for (redis_request.RedisRequestOuterClass.Command command :
+                requestBuilder.getTransaction().getCommandsList()) {
+            assertEquals(CustomCommand, command.getRequestType());
+            assertEquals("GETSTRING", command.getArgsArray().getArgs(0));
+            assertEquals(resultPayloads.pop(), command.getArgsArray().getArgs(1));
+        }
+    }
+
+    @SneakyThrows
+    @Test
+    public void submitNewCommand_with_Transaction_throws_interruptedException() {
+        // setup
+        String[] arg1 = new String[] {"GETSTRING", "one"};
+        String[] arg2 = new String[] {"GETSTRING", "two"};
+        String[] arg3 = new String[] {"GETSTRING", "two"};
+        Transaction trans =
+                new Transaction().customCommand(arg1).customCommand(arg2).customCommand(arg3);
+
+        CompletableFuture<Response> futureResponse = mock(CompletableFuture.class);
+        CompletableFuture<Object> futureObject = mock(CompletableFuture.class);
+        when(channelHandler.write(any(), anyBoolean())).thenReturn(futureResponse);
+        InterruptedException interruptedException = new InterruptedException();
```

Comment:
Test with another type of exception (e.g. `IllegalArgumentException`) and/or move to exception handling tests

Reply:
Test removed